src/java.desktop/share/classes/sun/java2d/pipe/RenderQueue.java

```diff
@@ -72,19 +75,28 @@
 public abstract class RenderQueue {
 
     /** The size of the underlying buffer, in bytes. */
-    private static final int BUFFER_SIZE = 32000;
+    private static final int BUFFER_SIZE;
+
+    static {
+        // Default 32K is too small for high-end GPU:
+        BUFFER_SIZE = align(getInteger("sun.java2d.render_queue.size", 32 * 1024, 32 * 1024, 16 * 1024 * 1024), 1024);
```

Comment:
Default value is 32768 instead of 32000 (aligned to 1024 bytes), so no "real" change to the current behaviour. If you prefer, I can use 32000 as before